sql/sql_window.cc

```diff
@@ -275,9 +275,12 @@ setup_windows(THD *thd, Ref_ptr_array ref_pointer_array, TABLE_LIST *tables,
       /*
         "The declared type of SK shall be numeric, datetime, or interval"
         we don't support datetime or interval, yet.
+	But it really only matters if the frame is bounded.
       */
       Item_result rtype= win_spec->order_list->first->item[0]->result_type();
-      if (rtype != REAL_RESULT && rtype != INT_RESULT && 
+      if (!win_spec->window_frame->top_bound->is_unbounded() &&
+          !win_spec->window_frame->bottom_bound->is_unbounded() &&
```

Comment:
I think the new condition fits better to the `if` above:
```
if (win_spec->window_frame && 
        win_spec->window_frame->units == Window_frame::UNITS_RANGE)
```
(given that we only touch the case when both window frames are unbounded).
This will however break `win.test` , but probably it can be amended?

Reply:
Done. By moving this there is now multiple ORDER BY items allowed. Added a test case to ensure that its ordering the window correctly.